lib/index.js

```diff
@@ -68,20 +68,28 @@ function minify(path, state, bindingOptions) {
 	const placeholder = uniqueId(quasis.join(''));
 	const tags = encapsulationGetTags(bindingOptions);
 
-	const minified = htmlMinifier.minify(tags.opening + quasis.join(placeholder) + tags.closing, state.opts.htmlMinifier);
-	if (!minified.startsWith(tags.opening) || !minified.endsWith(tags.closing)) {
-		throw path.buildCodeFrameError(majorDeleteError);
-	}
+	try {
+		const minified = htmlMinifier.minify(tags.opening + quasis.join(placeholder) + tags.closing, state.opts.htmlMinifier);
+		if (!minified.startsWith(tags.opening) || !minified.endsWith(tags.closing)) {
+			throw path.buildCodeFrameError(majorDeleteError);
+		}
 
-	const parts = minified.slice(tags.opening.length, -tags.closing.length || undefined).split(placeholder);
-	if (parts.length !== quasis.length) {
-		throw path.buildCodeFrameError(majorDeleteError);
-	}
+		const parts = minified.slice(tags.opening.length, -tags.closing.length || undefined).split(placeholder);
+		if (parts.length !== quasis.length) {
+			throw path.buildCodeFrameError(majorDeleteError);
+		}
 
-	parts.forEach((raw, i) => {
-		const args = cookRawQuasi(state.babel, raw);
-		template.get('quasis')[i].replaceWith(t.templateElement(args, i === parts.length - 1));
-	});
+		parts.forEach((raw, i) => {
```

Comment:
Potentially suppressing an error inside the `parts.forEach` may be dangerous?  What if we have a situation where `template.get('quasis')[0].replaceWith(...)` succeeds then `template.get('quasis')[1].replaceWith(...)` throws an error?  I think this case would be an unrecoverable error.

Maybe at the start of the `minify` function we should do `let {failOnError} = state;` then force it true before processing `parts.forEach`?  I'm not sure this scenario is even possible maybe I'm just being paranoid?

Reply:
What I've done now is just assign `parts` conditionally, and only join when they're assigned.